internal/bufferpool/bufferpool_test.go

```diff
@@ -36,10 +37,23 @@ func TestBuffers(t *testing.T) {
 				buf := Get()
 				assert.Zero(t, buf.Len(), "Expected truncated buffer")
 				assert.NotZero(t, buf.Cap(), "Expected non-zero capacity")
+
+				b := getRandBytes()
+				_, err := buf.Write(b)
+				assert.NoError(t, err, "Unexpected error from buffer.Write")
+
+				assert.Equal(t, buf.Len(), len(b), "Expected same buffer size")
+
 				Put(buf)
 			}
 			wg.Done()
 		}()
 	}
 	wg.Wait()
 }
+
+func getRandBytes() []byte {
+	b := make([]byte, rand.Intn(5000))
+	rand.Read(b)
```

Comment:
technically read can return an error that is being ignored here. might be cleaner to not use this at all